File: sdkwork-backend-sdk-csharp/Api/MusicApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class MusicApi
    {
        private readonly HttpClient _client;

        public MusicApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update music
        /// </summary>
        public async Task<PlusApiResultPlusMusicVO?> UpdateAsync(PlusMusicForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusMusicVO>(ApiPaths.BackendPath("/music"), body);
        }

        /// <summary>
        /// Create new music
        /// </summary>
        public async Task<PlusApiResultPlusMusicVO?> CreateAsync(PlusMusicForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusMusicVO>(ApiPaths.BackendPath("/music"), body);
        }

        /// <summary>
        /// Get music records by page
        /// </summary>
        public async Task<PlusApiResultPagePlusMusicVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusMusicVO>(ApiPaths.BackendPath("/music/list"), body, query);
        }

        /// <summary>
        /// Get all music records
        /// </summary>
        public async Task<PlusApiResultListPlusMusicVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusMusicVO>(ApiPaths.BackendPath("/music/list/all"), body);
        }

        /// <summary>
        /// Get music by ID
        /// </summary>
        public async Task<PlusApiResultPlusMusicVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusMusicVO>(ApiPaths.BackendPath($"/music/{id}"));
        }

        /// <summary>
        /// Delete music
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/music/{id}"));
        }
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/MusicApi.swift
import Foundation

public class MusicApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update music
    public func update(body: PlusMusicForm) async throws -> PlusApiResultPlusMusicVO? {
        let response = try await client.put(ApiPaths.backendPath("/music"), body: body)
        return response as? PlusApiResultPlusMusicVO
    }

    /// Create new music
    public func create(body: PlusMusicForm) async throws -> PlusApiResultPlusMusicVO? {
        let response = try await client.post(ApiPaths.backendPath("/music"), body: body)
        return response as? PlusApiResultPlusMusicVO
    }

    /// Get music records by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusMusicVO? {
        let response = try await client.post(ApiPaths.backendPath("/music/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusMusicVO
    }

    /// Get all music records
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusMusicVO? {
        let response = try await client.post(ApiPaths.backendPath("/music/list/all"), body: body)
        return response as? PlusApiResultListPlusMusicVO
    }

    /// Get music by ID
    public func getById(id: String) async throws -> PlusApiResultPlusMusicVO? {
        let response = try await client.get(ApiPaths.backendPath("/music/\(id)"))
        return response as? PlusApiResultPlusMusicVO
    }

    /// Delete music
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/music/\(id)"))
        return response as? PlusApiResultBoolean
    }
}
